lib/parser.dart

```diff
@@ -313,7 +313,15 @@ class HtmlParser {
       var token = tokenizer.current;
       var newToken = token;
       int type;
+      var attemptCount = 0;
       while (newToken != null) {
+        // Limit the number of attempts to process the current token, and give
```

Comment:
Do we know why this loop gets stuck?

Reply:
The `endTagOther` method [0] is called from the `startTagButton` [1]:
[0]: https://github.com/dart-lang/html/blob/master/lib/parser.dart#L2247-L2268
[1]: https://github.com/dart-lang/html/blob/master/lib/parser.dart#L1735

It adds an error, does not changes the state, and returns the same token that is getting retried in the loop.

Reply:
Can we add a comment here:
` // HACK: Downgrade infinite OOM bug to an error.`